main.nf

```diff
@@ -1,167 +1,57 @@
-    input:
-      tuple val(repo_url), val(repo_name), path(repo_dir), val(out_dir), path("status_repo.txt")
-    output:
-      // Emit a tuple: (repo_url, repo_name, out_dir, file(status_almanack_<repo_name>.txt))
-      tuple val(repo_url), val(repo_name), val(out_dir), file("status_almanack_${repo_name}.txt")
-    script:
-    """
-    mkdir -p ${out_dir}
-    # Copy the repository to /tmp for faster I/O
-    cp -r ${repo_dir} /tmp/repo
-    # Run Almanack analysis; if it completes without error, mark as PASS
-    if python3 -c "import json, almanack; print(json.dumps(almanack.table(repo_path='/tmp/repo')))" > ${out_dir}/almanack-results.json 2>/dev/null; then
-         ALMANACK_STATUS="PASS"
-    else
-         ALMANACK_STATUS="FAIL"
-    fi
-    # Append Almanack status to the previous summary line from status_repo.txt and write to a unique file
-    PREV_STATUS=\$(cat status_repo.txt)
-    echo "\${PREV_STATUS},\${ALMANACK_STATUS}" > status_almanack_${repo_name}.txt
-    """
-}
-
-/*-----------------------------------------------
-   Process: GenerateReport
-   - Aggregates all uniquely named status_almanack files into one CSV file.
------------------------------------------------*/
-process GenerateReport {
-    errorStrategy 'ignore'
-    input:
-      // Collect all status files (with names matching the pattern)
-      file(status_files)
-    output:
-      file "consolidated_report.csv"
-    script:
-    """
-    # Write header
-    echo "Tool,CloneRepository,CheckReadme,CheckDependencies,CheckTests,Almanack" > consolidated_report.csv
-    # Append each status row from all files
-    for file in ${status_files}; do
-      cat \$file >> consolidated_report.csv
-    done
-    """
-}
-
-/*-----------------------------------------------
-   Workflow Section
-   - Creates a channel from a sample sheet (or single repo_url)
-   - Maps each repository URL to a tuple: (repo_url, repo_name, out_dir)
-   - Pipes these tuples through ProcessRepo and RunAlmanack.
-   - Then collects all unique status files into one list and sends them to GenerateReport.
------------------------------------------------*/
 workflow {
-    // Build a channel from either a sample sheet or a single repo URL.
-    def repoCh = params.sample_sheet ? 
-         Channel.fromPath(params.sample_sheet)
-                .splitCsv(header:true)
-                .map { row -> row.repo_url } :
-         ( params.repo_url ? Channel.value(params.repo_url) : error("You must provide either a sample_sheet or a repo_url parameter.") )
+    // Build a channel from either a sample sheet or a single repo URL
+    def repoCh
+    if (params.sample_sheet) {
+        repoCh = Channel.fromPath(params.sample_sheet)
+                        .splitCsv(header:true)
+                        .map { row -> row.repo_url }
+    } else if (params.repo_url) {
+        repoCh = Channel.value(params.repo_url)
+    } else {
+        error "Provide either a sample_sheet or repo_url."
+    }
```

Comment:
No immediate change needed but consider behavior if both sample_sheet and repo_url params are provided